### PR TITLE
Fix Clippy 1.75.0 errors

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: stratis-storage/project#682

### CODE REVIEW

File: src/engine/structures/lock.rs [hunk 1]
```diff
@@ -370,6 +370,7 @@ where
     /// Returns true if:
     /// * The waiting task does not conflict with any already woken tasks.
     /// * The waiting task does not conflict with any locks currently held.
+    #[allow(clippy::get_first)]
```
Comment (mulkieran) [inline src/engine/structures/lock.rs:373]: Instead of the allow, use ```front()``` for ```get(0)```.

**Review by mulkieran — CHANGES_REQUESTED** [review]
One request...

File: src/engine/structures/lock.rs (hunk 1 above)
Comment (bgurney-rh) [reply src/engine/structures/lock.rs:373]: OK; now I see it.  It turns out that clippy did recommend `front()`, but it was in a large list of `first()` recommendations.  I'll revert the clippy allow, let this change run through CI, and then rebase to remove the unnecessary allow commit if all tests pass.

**Review by mulkieran — APPROVED** [review]

**Comment by mulkieran** [comment]
> Cockpit tests failed for commit [3849d76](https://github.com/stratis-storage/stratisd/commit/3849d769f2c9e5923253bb7c55070c6d2ae3844b). @martinpitt, @jelly, @mvollmer please check.

These are just minor lint fixes, so likely the failure is not due to us.

**Comment by bgurney-rh** [comment]
Rebasing to remove the commits that added and reverted `Allow clippy::get_first...`

**Comment by mulkieran** [comment]
Merging. I think there is something wrong with the testing framework on rawhide.

**Comment by martinpitt** [comment]
@mulkieran Right, something broke in the storage stack in rawhide yesterday. It's already on our large pile of stuff to look at which broke over the EOY holidays. Sorry for the trouble, please ignore for now -- tests passed on F39, which gives sufficient confidence. Your own `tests.udev.test_udev.UdevTest1.test_generic` failed as well, does that talk to only udev, or also udisks/stratisd etc?

**Comment by mulkieran** [comment]
That test is a test that essentially test stratisd's correct response to a storm of udev events. The tests are unfortunately a bit flaky, but also too useful for us to abandon.

> @mulkieran Right, something broke in the storage stack in rawhide yesterday. It's already on our large pile of stuff to look at which broke over the EOY holidays. Sorry for the trouble, please ignore for now -- tests passed on F39, which gives sufficient confidence. Your own `tests.udev.test_udev.UdevTest1.test_generic` failed as well, does that talk to only udev, or also udisks/stratisd etc?

**Pull request merged**